### PR TITLE
KTOR-2709 Use kotlin.reflect.javaType instead of type token

### PR DESCRIPTION
**Pull request by cliffred**
**Subsystem**
JVM Utils

**Motivation**
Fix https://youtrack.jetbrains.com/issue/KTOR-2709
Using kotlin.reflect.javaType doesn't produce an anonymous class for each call site. Also, this prevents Ktor from being affected by this compiler bug: https://youtrack.jetbrains.com/issue/KT-46797.

**Solution**
Use kotlin.reflect.javaType instead of type token pattern.

### CODE REVIEW

**Comment by rsinukov** [comment]
Hi, thanks for the PR!
I will run the tests to see if we are not affected by any corner cases.

File: ktor-utils/jvm/src/io/ktor/util/reflect/TypeInfoJvm.kt [hunk 1]
```diff
@@ -4,22 +4,15 @@
...
 package io.ktor.util.reflect
 
-import java.lang.reflect.*
-import java.lang.reflect.Type
 import kotlin.reflect.*
 
 public actual typealias Type = java.lang.reflect.Type
 
-@PublishedApi
-internal open class TypeBase<T>
```
Comment (rsinukov) [inline ktor-utils/jvm/src/io/ktor/util/reflect/TypeInfoJvm.kt:14]: Since this class is annotated with `@PublishedApi`, it's part of the public API and can't be removed in minor release. Please keep it, but add a deprecation to it.

**Review by rsinukov — CHANGES_REQUESTED** [review]
LGTM in general and tests look good, but small comment

**Review by rsinukov — APPROVED** [review]
LGTM
